src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+# encoding: utf-8
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+#
+# Module: Manipulate system authentication configuration.
+# Author: Howard Guo <hguo@suse.com>
+
+require 'yast'
+require 'resolv'
+require 'open3'
+require 'socket'
+require 'tempfile'
+require 'fileutils'
+require 'date'
+
+module Auth
+    class AuthConf
+        include Yast
+        include I18n
+        include Logger
+        include UIShortcuts
+
+        # Return network host name and environment facts.
+        def self.get_net_facts
+            resolvable_name = ''
+            begin
+                resolvable_name = Socket.gethostbyname(Socket.gethostname)
+                resolvable_name = resolvable_name.fetch(0, '')
+            rescue
```

Comment:
generic rescue without specific type is discouraged practice.
E.g. consider that fetch failed. In such case resolvable name is ` Socket.gethostbyname(Socket.gethostname)` object. which probably cause strange failure later.